### PR TITLE
feat: save some memory allocations on un-used cache store

### PR DESCRIPTION
**Pull request by yihuang**
# Description

Closes: #XXXX

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [ ] added a changelog entry to `CHANGELOG.md`
* [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (bank) [#237](https://github.com/crypto-org-chain/cosmos-sdk/pull/237) Support virtual accounts in sending coins.
 * (x/bank) [#239](https://github.com/crypto-org-chain/cosmos-sdk/pull/239) Add low level `AddBalance`,`SubBalance` APIs to bank keeper.
 * [#243](https://github.com/crypto-org-chain/cosmos-sdk/pull/243) Support `RunAtomic` API in `Context` to use new CoW branched cache store.
+* [#244](https://github.com/crypto-org-chain/cosmos-sdk/pull/244) Init btree store lazily to save allocations when no content insert into it.
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
* [#248](https://github.com/crypto-org-chain/cosmos-sdk/pull/248) Init btree store lazily to save allocations when no content insert into it.
```

File: store/internal/btree/btree.go [hunk 2]
```diff
@@ -28,25 +28,36 @@ type BTree[V any] struct {
...
+		var zero V
+		return zero, false
+	}
 	i, found := bt.tree.Get(newItemWithKey[V](key))
 	return i.value, found
 }
 
-func (bt BTree[V]) Delete(key []byte) {
+func (bt *BTree[V]) Delete(key []byte) {
+	bt.init()
```
Comment (mmsqe) [inline store/internal/btree/btree.go]: do we just return if bt.tree is nil here

Comment (yihuang) [reply store/internal/btree/btree.go]: good catch👍

Comment (yihuang) [inline store/internal/btree/btree.go]:
```suggestion
	if bt.tree == nil {
		return
	}
```

**Review by mmsqe — APPROVED** [review]

**Pull request merged**